client/jetpack-cloud/sections/partner-portal/hooks.tsx

```diff
@@ -514,3 +514,73 @@ export function useIssueMultipleLicenses(
 
 	return [ issue, isLoading ];
 }
+
+/**
+ * Handle multiple license assignment
+ *
+ */
+export function useAssignMultipleLicenses(
+	selectedLicenseKeys: Array< string >,
+	selectedSite: { ID: number; domain: string } | null
+): [ () => void, boolean ] {
+	const products = useProductsQuery( {
+		select: selectAlphaticallySortedProductOptions,
+	} );
+	const dispatch = useDispatch();
+	const assignLicense = useAssignLicenseMutation( {
+		onSuccess: () => {
+			page.redirect( partnerPortalBasePath( '/licenses' ) );
+		},
+		onError: () => {
+			page.redirect( partnerPortalBasePath( '/licenses' ) );
+		},
+	} );
+	const isLoading = assignLicense.isLoading;
+	const selectedSiteId = selectedSite?.ID as number;
+	const assign = useCallback( async () => {
+		// setIsSubmitting( true );
```

Comment:
I feel like the whole code written in this function can be moved into a different reusable function and used here and in `useIssueMultipleLicenses` since we are doing the same thing there. Let me know your thoughts.